config/serverless.yml

```diff
@@ -70,6 +70,9 @@ core.lifecycle.disablePreboot: true
 # Enable ZDT migration algorithm
 migrations.algorithm: zdt
 
+# Enable elasticsearch response size circuit breaker
+elasticsearch.maxResponseSize: "100mb"
```

Comment:
Nit: Similar to the dns cache, this one feels like something we'd like to try first in QA before extending it to prod. At the same time, having it here will likely catch things in our CI (release process).

Reply:
From our preliminary investigations, 100mb responses are very unfrequent, and every time we encountered one, it caused an OOM terminating the process, but I feel like having this value here for all environments from the start is probably ok, but I'll defer to @jloleysens.

> My main concern is around streamed responses... in those cases we might be fine with allowing any size, right?

It depends. If the response is then streamed back to the client, it likely is. If the stream is fully consumed on the server, it doesn't change much.

But I agree that this is something we can improve later, as atm there's exactly 2 usages of `asStream` for the client (one from a `map` route for tiles, one for the `files` plugin). One option could be to unset the limit parameters when the `asStream` option is set, from our Transport class (similar to what I did to keep the two options in sync)

Reply:
Over the last ~30d we've had 4 large responses so I agree it is low. However, this setting applies to all requests to ES (similar to DNS caching) while arguably a bit less sensitive. Same as DNS, IMO that means: watch the next release closely or rollout incrementally (as @afharo did with DNS).

Either way we should have some monitoring (even if adhoc) for when/if this response size limit is exceeded.